File: algorithm/src/main/java/baekjoon/p05426/Main.java
package baekjoon.p05426;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        while (t-- > 0) {
            System.out.println(solve(sc.next()));
        }
    }

    private static String solve(String input) {
        int len = (int) Math.sqrt(input.length());
        char[][] a = new char[len][len];
        char[][] b = new char[len][len];

        int idx = 0;
        for (int i = 0; i < input.length(); i += len) {
            a[idx++] = input.substring(i, i + len).toCharArray();
        }

        for (int i = 0; i < len; i++) {
            char[] arr = new char[len];
            for (int j = 0; j < arr.length; j++) {
                arr[j] = a[j][len - i - 1];
            }
            b[i] = arr;
        }

        StringBuilder sb = new StringBuilder();
        for (char[] e : b) {
            sb.append(e);
        }

        return sb.toString();
    }

}

File: algorithm/src/main/java/baekjoon/p05426/Main.kt
package baekjoon.p05426

import kotlin.math.sqrt

fun main(args: Array<String>) {
    repeat(readLine()!!.toInt()) {
        println(solve(readLine()!!))
    }
}

private fun solve(input: String): String {
    val len = sqrt(input.length.toDouble()).toInt()
    val a = Array(len) { CharArray(len) }
    val b = Array(len) { CharArray(len) }

    var si = 0
    var ai = 0
    while (si < input.length) {
        a[ai] = input.substring(si, si + len).toCharArray()
        si += len
        ai++
    }

    for (i in 0 until len) {
        b[i] = (0 until len).map { a[it][len - i - 1] }.toCharArray()
    }

    return buildString {
        for (e in b) {
            append(e)
        }
    }
}